calitp/storage.py

```diff
@@ -473,6 +405,63 @@ def get_latest_file(
     return ret
 
 
+def get_latest(
+    cls: Type[PartitionedGCSArtifact],
+    bucket: str = None,
+    table: str = None,
+    partition_names: List[str] = None,
+) -> PartitionedGCSArtifact:
+    if not bucket:
+        bucket = cls.bucket
+
+        if not isinstance(bucket, str):
+            raise TypeError(f"must either pass bucket, or the bucket must resolve to a string; got {type(bucket)}")
+
+    if not table:
+        table = cls.table
+
+        if not isinstance(table, str):
+            raise TypeError(f"must either pass table, or the table must resolve to a string; got {type(table)}")
+
+    if not partition_names:
+        partition_names = cls.partition_names
+
+        if not isinstance(partition_names, list):
+            raise TypeError(
+                f"must either pass partition names, or the partition names must resolve to a list; got {type(partition_names)}"
+            )
+
+    latest = get_latest_file(
+        bucket,
+        table,
+        prefix_partitions={},
+        # TODO: this doesn't pick up the type hint of dt since it's a property; it's fine as a string but we should fix
+        partition_types={name: get_type_hints(cls).get(name, str) for name in partition_names},
+    )
+
+    logging.info(f"identified {latest.name} as the most recent extract of {cls}")
+
+    return cls(**json.loads(get_fs().getxattr(path=f"gs://{latest.name}", attr=PARTITIONED_ARTIFACT_METADATA_KEY)))
+
+
+class AirtableGTFSDataRecord(BaseModel):
+    id: str
+    name: str
+    uri: Optional[str]
+    pipeline_url: Optional[str]
+    data: Optional[str]
+    data_quality_pipeline: Optional[bool]
+    schedule_to_use_for_rt_validation: Optional[List[str]]
+    authorization_url_parameter_name: Optional[str]
+    url_secret_key_name: Optional[str]
+    authorization_header_parameter_name: Optional[str]
+    header_secret_key_name: Optional[str]
+    auth_query_param: Dict[str, str] = {}
```

Comment:
sorry, why do we still have `auth_query_param` here but not an equivalent for headers?

Reply:
Because these were split off and saved as a field from the uri on parse, while headers were entirely derived from the uri as a property. We do need this still to help in the migration.